### DISCUSSION TITLE
Disable automatic schema refetching when switching to another saved request

### DISCUSSION THREAD

**Discussion by martasd** [Kong/insomnia#3543] [Ideas]
**Is your feature request related to a problem? Please describe.**

I have saved a number of requests against my API:

![image (11)](https://user-images.githubusercontent.com/591911/87543968-f1643080-c6a5-11ea-9b2e-3c311d799660.png)

Every time I switch to a different request in my workspace the schema gets fetched from the server. The majority of cases the schema does not change, so there is no need to refetch it spending extra time and resources doing so. This happens despite having **Automatic Fetch** disabled and despite using **the same URL** for each request:

![image (12)](https://user-images.githubusercontent.com/591911/87544029-0e006880-c6a6-11ea-82a1-b25d661eea3f.png)

The pain point is that in order to be able to access the schema documentation, I need to wait (in my case about 10s) for the schema to be fetched every time I switch to a different request. This is very uncomfortable as I switch to various requests 20 or 30 times a day.

**Describe the solution you'd like**

I would like to be able to store the schema so that it could be used for multiple defined requests without being refetched. In other words, I would like to be able to disable the automatic fetching of schema when I switch to a different saved request.

**Describe alternatives you've considered**

I would love to be able to use a workaround, but I have not found any so far.

**Comment by martasd** [comment]
Hi @nijikokun. Does this feature request make sense? Do you know if there's any workaround to achieve this?

**Comment by eamigo86** [comment]
Hi @nijikokun, I have the same problem from some time ago and it's really painful to work like this, sometimes I needed to change to Playground to work faster.

__Describe the solution you'd like:__
Ideally, if the last URL the introspection call was made from could be saved or cached for some time (maybe ~1 hour), and if the user switches to another request in Insomnia and the graphql URL is the same, then schema will not be updated automatically. 
Right now, the user always could force the schema update manually if he wants from the "Refresh Schema" option.

**Comment by eamigo86** [comment]
Hi @nijikokun, any progress with this issue? Still occur in the latest version (v2021.05.0)

Hi @martasd, do you found some solution for this?

**Reply by martasd** [reply]
Hi @eamigo86. No, I have not found a solution. It looks like this is still an issue in Insomnia. Fortunately for me, I now work on a project where schema fetching is a lot faster (under 1s), so this is not painful for me right now.

**Comment by dimitropoulos** [comment]
hey @eamigo86, sorry this has gone so long without response.  I've asked around internally on this and I hope we'll have some path forward for you sooner rather than later.  Sorry I don't have anything more specific to say at this moment but I wanted to at least let you know you've been heard.  We're trying to improve our responsiveness to github issues and github discussions like this.

**Reply by eamigo86** [reply]
Hi @dimitropoulos, any progress with this issue?

**Reply by dimitropoulos** [reply]
none yet, but I followed up again just now and we'll see what comes of that.  while we do have some high priority other things in the queue, I would like to think that we can find time to do this one since the scope of it is quite small.

**Reply by eamigo86** [reply]
Thank you very much for answering @dimitropoulos. The truth is that this problem has been reported and opened for a long time. Right now I am using **Altair GraphQL Client** since it doesn't have this problem and it is specialized in Graphql
